### DISCUSSION TITLE
IAND and IOR instructions added (and now all operations)

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#160] [Progress Updates]
for the data conversions D2X, etc. we need bitwise operations. IAND/IOR. They operate on 2 integers resulting in a new integer:
IAND r1,r2,r3   or  IAND r2,r2,integer
IOR   r1,r2,r3   or   IOR    r2,r2,integer

**Comment by Peter-Jacob** [comment]
what about xor, complement, and shift bit operations?

**Comment by Peter-Jacob** [comment]
I took the liberty to complete all bit operations: 

```
instruction names
-------------------
iand  AND operation of 2 integers
ior    OR  operation of 2 integers
ixor  XOR operation of 2 integers
ishl  shift logical left of an integer
ishr  shift logical right of an integer
inot  invert bits of an integer (also known as one-complement)

```
I added where possible  2 versions reg,reg,reg and reg,reg,int. We can reduce instruction numbers if we just use the reg,reg,reg version. Please advice.

**Reply by Peter-Jacob** [reply]
There is no urgency to add them to the compiler, if at all. If necessary I use them as assembler statements within my level b rexxes.

**Reply by adesutherland** [reply]
Amazing! We do need to decide how (if) we are going to extend the REXX operators - also considering the C standards etc etc. I don't want to rush this, if that is OK ...

**Comment by rvjansen** [comment]
Excellent!

René.
